main.go

```diff
@@ -1,25 +1,23 @@
 package main
 
 import (
-	"fmt"
 	"os"
 	"path/filepath"
 
-	"github.com/hashicorp/go-hclog"
-	goplugin "github.com/hashicorp/go-plugin"
-	"github.com/hashicorp/terraform/plugin"
-	"github.com/hashicorp/terraform/plugin/discovery"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
 func main() {
-	provider := getInstanceOfAzureRMProvider()
+	provider := getInstanceOfProvider("azurerm")
```

Comment:
Would it make sense for the provider name come from an environment variable or something like that?

Reply:
Yes, I'm going to pick this up as #15